hollow/src/main/java/com/netflix/hollow/api/producer/HollowProducer.java

```diff
@@ -428,12 +428,13 @@ protected void runCycle(Populator task, ProducerStatus.Builder cycleStatus, long
         try {
             // 1a. Prepare the write state
             writeEngine.prepareForNextCycle();
-            WriteState writeState = new WriteStateImpl(toVersion, objectMapper, readStates.current());
+            WriteStateImpl writeState = new WriteStateImpl(toVersion, objectMapper, readStates.current());
 
             // 2. Populate the state
             ProducerStatus.Builder populateStatus = listeners.firePopulateStart(toVersion);
             try {
                 task.populate(writeState);
+                writeState.seal();
```

Comment:
Should be performed in a finally block.

I would argue that the complete WriteState instance should become inoperative if the instance is operated on beyond the scope of population stage (that includes multiple threads and/or the current thread if the instance leaks out). That sends a very clear message on the behavior and is easier to specify. In this respect the instance is not sealed but made inoperable.

Reply:
> Should be performed in a finally block.

Good catch. Fixed.

> I would argue that the complete WriteState instance should become inoperative...

So all public methods would start throwing? And we either expose package-private accessors *or* stop using `WriteState` internally to pass things around?

The current implementation of `HollowIncrementalProducer` relies on `getStateEngine()`, but I'm not opposed to spending a little bit of time to make the incremental producer work with whatever changes we want to make to `HollowProducer`.